exasol_sphinx_github_pages_generator/deploy_github_pages.py

```diff
@@ -90,43 +90,53 @@ def build_and_copy_documentation(build_dir, worktree, source_branch, source_dir)
         shutil.rmtree(output_dir)
     print(f"(Re)Creating output directory {output_dir}")
     output_dir.mkdir(parents=True)
-    print(f"Copying HTML output {html_output_dir} to the output directory {output_dir}")
-    # TODO find "$HTML_OUTPUT_DIR" -mindepth 1 -maxdepth 1 -exec mv -t "$OUTPUT_DIR" -- {} +
+    print(f"Copying HTML output {build_dir} to the output directory {output_dir}")
+    for obj in os.listdir(build_dir):
+        shutil.move(build_dir + "/" + str(obj), output_dir)
+    # TODO test correctness of : find "build_dir" -mindepth 1 -maxdepth 1 -exec mv -t "$OUTPUT_DIR" -- {} +
     print(f"Content of output directory {output_dir}")
-
-    # touch "$WORKTREE/.nojekyll"
-    # run([ls -la "$OUTPUT_DIR"])
+    open(f"{worktree}/.nojekyll", "w").close()
+    run(["ls", "-la", output_dir])
     return output_dir
 
-def git_commit_and_push(push_origin, push_enabled, source_branch, output_dir, current_commit_id, target_branch):
-    #pushd "$WORKTREE"
-    print("Current directory before commit and push $PWD") #TODO
+#todo remove debug output
+def git_commit_and_push(worktree, push_origin, push_enabled, source_branch, output_dir, current_commit_id, target_branch):
+    currentworkdir = os.getcwd()
+    print(currentworkdir)
+    os.chdir(worktree)
+    print(f"Current directory before commit and push {os.getcwd()}")
     print("Git commit")
-    with open (f"{output_dir}/.source", "w+") as file:
-        file.write(f"BRANCH={source_branch}")
+    with open(f"{output_dir}/.source", "w+") as file:
+        file.write(f"BRANCH={source_branch} \n")
         file.write(f"COMMIT_ID={current_commit_id}")
-    run(["git", "add", "."])
-    run(["git", "diff-index", "--quiet", "HEAD", "--", "||", "git", "commit", "--no-verify", "-m", f"Update documentation from source branch {source_branch} with commit id {current_commit_id}"])
+    run(["git", "add", "-v", "."])
+    run(["git", "status"])
+    changes_exists = run(["git", "diff-index", "--quiet", "HEAD", "--"], capture_output=True, text=True)
+    if 1 == changes_exists.returncode:
+        print(f"committing changes because changes_exist is {changes_exists.returncode}")
+        run(["git", "commit", "--no-verify", "-m", f"Update documentation from source branch {source_branch} with commit id {current_commit_id}"])
     if push_origin != "" and push_enabled == "push":
         print(f"Git push {push_origin} {target_branch}")
         run(["git", "push", push_origin, target_branch])
-    #popd
+    os.chdir(currentworkdir)
```

Comment:
1. Do we need to push of there is no change (if we didn't commit)?
2. Maybe we should print a log if we there was no change (change_exists != 1)

Reply:
done